azurelinuxagent/ga/exthandlers.py

```diff
@@ -1313,6 +1314,16 @@ def download(self):
 
         self.pkg_file = destination
 
+    def ensure_consistent_data_for_mc(self):
+        # If CRP expects Handler to support MC, ensure the HandlerManifest also reflects that.
+        # Even though the HandlerManifest.json is not expected to change once the extension is installed,
+        # CRP can wrongfully request send a Multi-Config GoalState even if the Handler supports only Single Config.
+        # Checking this only if HandlerState == Enable. In case of Uninstall, we dont care.
+        if self.supports_multi_config and not self.load_manifest().supports_multiple_extensions():
+            raise ExtensionConfigError(
```

Comment:
this will propagate to telemetry, right? (i.e. not only to the local log) thanks

Reply:
Yup yup, will be reported to CRP and sent to Kusto too - 

>        except ExtensionConfigError as error:
            # Catch and report Invalid ExtensionConfig errors here to fail fast rather than timing out after 90 min
            err_msg = "Ran into config errors: {0}. \nPlease retry again as another operation with updated settings".format(
                ustr(error))
            self.__handle_and_report_ext_handler_errors(ext_handler_i, error,
                                                        report_op=WALAEventOperation.InvalidExtensionConfig,
                                                        message=err_msg, extension=extension)